SharpPcap/LibPcap/LibPcapLiveDevice.cs

```diff
@@ -156,67 +180,89 @@ public override void Open(DeviceConfiguration configuration)
-                }
+                    Configure(
+                        configuration, nameof(configuration.Monitor),
+                        LibPcapSafeNativeMethods.pcap_set_rfmon, (int?)configuration.Monitor
+                    );
 
-                Configure(
-                    configuration, nameof(configuration.Monitor),
-                    LibPcapSafeNativeMethods.pcap_set_rfmon, (int?)configuration.Monitor
-                );
+                    Configure(
+                        configuration, nameof(configuration.BufferSize),
+                        LibPcapSafeNativeMethods.pcap_set_buffer_size, configuration.BufferSize
+                    );
 
-                Configure(
-                    configuration, nameof(configuration.BufferSize),
-                    LibPcapSafeNativeMethods.pcap_set_buffer_size, configuration.BufferSize
-                );
 
-                // Check if immediate is supported
-                var immediate_supported = Pcap.LibpcapVersion >= new Version(1, 5, 0);
-                // Check if we can do immediate by setting mintocopy to 0
-                // See https://www.tcpdump.org/manpages/pcap_set_immediate_mode.3pcap.html
-                var mintocopy_supported = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+                    if (configuredImmediate.HasValue)
+                    {
+                        if (!immediate_supported && !mintocopy_supported)
+                        {
+                            configuration.RaiseConfigurationFailed(
+                                nameof(configuredImmediate),
+                                (int)PcapError.PlatformNotSupported
+                            );
+                        }
+                        else if (immediate_supported)
+                        {
+                            var immediate = configuredImmediate.Value ? 1 : 0;
+                            Configure(
+                                configuration, nameof(configuredImmediate),
+                                LibPcapSafeNativeMethods.pcap_set_immediate_mode, immediate
+                            );
+                        }
+                    }
 
-                if (configuration.Immediate.HasValue)
+                    var activationResult = LibPcapSafeNativeMethods.pcap_activate(Handle);
+                    if (activationResult < 0)
+                    {
+                        string err = "Unable to activate the adapter (" + Name + "). Return code: " + activationResult.ToString();
+                        throw new PcapException(err);
+                    }
+                }
+                else
                 {
-                    if (!immediate_supported && !mintocopy_supported)
+                    // Shall we silently ignore unsupported configuration parameters
+                    // TimestampResolution, TimestampType, Monitor and BufferSize
+                    // or check and throw exception?
```

Comment:
Call `configuration.RaiseConfigurationFailed`, that method shall decide to throw or ignore.